src/System.Memory/src/System/Buffers/ReadOnlySequence_helpers.cs

```diff
@@ -188,91 +192,147 @@ private static SequencePosition SeekMultiSegment(ReadOnlySequenceSegment<T> curr
             }
 
             // Hit the end of the segments but didn't reach the count
-            if (currentSegment == null || (currentSegment == endObject && endPosition < offset))
+            if (currentSegment == null || endIndex < offset)
                 ThrowHelper.ThrowArgumentOutOfRangeException_OffsetOutOfRange();
 
         FoundSegment:
             return new SequencePosition(currentSegment, (int)offset);
         }
 
-        private static void CheckEndReachable(object startSegment, object endSegment)
+        private void CheckEndReachable(ReadOnlySequenceSegment<T> current)
```

Comment:
Do remove this method?

Reply:
It is currently being used. So we can't remove it.